TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/AbstractNodeData.php

```diff
@@ -213,7 +213,7 @@ public function hasProperty($propertyName)
         if (is_object($this->contentObjectProxy)) {
             return ObjectAccess::isPropertyGettable($this->contentObjectProxy->getObject(), $propertyName);
         }
-        return isset($this->properties[$propertyName]);
+        return isset($this->properties[$propertyName]) || array_key_exists($propertyName, $this->properties);
```

Comment:
How much faster (in percent as well as in practice)? If not really substantially faster, I'd avoid those (micro-)optimisations and favour straightforward code.